### PR TITLE
subsys: sdfw_builtin: Remove legacy implementation

### PR DESCRIPTION
**Pull request by tomchy**
Remove legacy implementation of SDFW builtin keys.

Ref: NCSDK-28698

### CODE REVIEW

**Review by kszromek-nordic — APPROVED** [review]

**Review by adsz-nordic — APPROVED** [review]

**Review by robertstypa — APPROVED** [review]

**Review by ahasztag — APPROVED** [review]
I haven't found any issues with this commit

File: subsys/suit/platform/sdfw/src/suit_plat_authenticate.c [hunk 1]
```diff
@@ -10,18 +10,61 @@
...
+#define MANIFEST_PUBKEY_GEN_RANGE	 2
+
+/** @brief Convert int-based key ID into structure that holds both the ID and the key owner.
+ *
+ * @param[out]  key     Key structure to set.
+ * @param[in]   key_id  Int-based key ID.
+ *
+ * @returns SUIT_PLAT_SUCCESS  In case of successful conversion, SUIT_PLAT_ERR_INVAL otherwise.
+ */
+static suit_plat_err_t convert_int_key(mbedtls_svc_key_id_t *key, uint32_t key_id)
```
Comment (vili-nordic) [inline subsys/suit/platform/sdfw/src/suit_plat_authenticate.c:31]: You don't need to this. We want the source of request to be encoded as the owner, and then we will verify that the access is allowed. Secure domain is allowed to access all keys.

Comment (tomchy) [reply subsys/suit/platform/sdfw/src/suit_plat_authenticate.c:31]: Please agree on the common version with @osaether .
If that's the case, we can use a solution based on: https://github.com/nrfconnect/sdk-nrf/commit/4685d64f7026f261a25ee9102c8fc84dedcbd5b5

**Review by vili-nordic — APPROVED** [review]

**Comment by tomchy** [comment]
Temporarily rebased on https://github.com/nrfconnect/sdk-nrf/pull/16914 to run the CI with all fixes.

**Review by osaether — APPROVED** [review]
LGTM

**Pull request merged**